hubspot/parse.go

```diff
@@ -76,3 +77,72 @@ func parsePagingNext(node *ajson.Node) (*ajson.Node, error) {
 
 	return next, nil
 }
+
+// getRecords returns the records from the response.
+func getRecords(node *ajson.Node) ([]map[string]interface{}, error) {
+	records, err := node.GetKey("results")
+	if err != nil {
+		return nil, err
+	}
+
+	if !records.IsArray() {
+		return nil, ErrNotArray
+	}
+
+	arr := records.MustArray()
+
+	out := make([]map[string]interface{}, 0, len(arr))
+
+	for _, v := range arr {
+		if !v.IsObject() {
```

Comment:
Optional: It might be more convenient if you used GetObject (https://pkg.go.dev/github.com/spyzhov/ajson#Node.GetObject), and modified the return value of this function to `map[string]*Node`, then `getStructuredData` has can use ajson functions (e.g. GetKey to retrieve the value at `properties`). For future providers, some of them might require more complicated logic in their `getStructuredData` functions, so still having the data in ajson.Node format still will allow ajson functions to be used.

Reply:
Good idea, but I'll follow up with this in a separate PR for a review